net/HttpRequest.hpp

```diff
@@ -971,6 +978,7 @@ class Session final : public ProtocolHandlerInterface
     std::chrono::microseconds getTimeout() const { return _timeout; }
 
     std::shared_ptr<const Response> response() const { return _response; }
+    const std::string getUrl() const { return _request.getUrl(); }
```

Comment:
Why not return a reference? After all, that's what `_request.getUrl()` returns (so returning a copy always costs an allocation, even when not needed).